dask/array/core.py

```diff
@@ -2414,38 +2414,70 @@ def store(out, x, index, lock, region):
     return dsk
 
 
-def asarray(array):
-    """Coerce argument into a dask array
+def asarray(a):
+    """Convert the input to a dask array.
+
+    Parameters
+    ----------
+    a : array-like
+        Input data, in any form that can be converted to a dask array.
+
+    Returns
+    -------
+    out : dask array
+        Dask array interpretation of a.
 
     Examples
     --------
+    >>> import dask.array as da
+    >>> import numpy as np
     >>> x = np.arange(3)
-    >>> asarray(x)
+    >>> da.asarray(x)
     dask.array<array, shape=(3,), dtype=int64, chunksize=(3,)>
+
+    >>> y = [[1, 2, 3], [4, 5, 6]]
+    >>> da.asarray(y)
+    dask.array<array, shape=(2, 3), dtype=int64, chunksize=(2, 3)>
     """
-    if isinstance(array, Array):
-        return array
-    if not isinstance(getattr(array, 'shape', None), Iterable):
-        array = np.asarray(array)
-    return from_array(array, chunks=array.shape, getitem=getter_inline)
+    if isinstance(a, Array):
+        return a
+    if not isinstance(getattr(a, 'shape', None), Iterable):
+        a = np.asarray(a)
```

Comment:
If this ends up being a list of Dask Arrays, this could be problematic. FWIW this can be done with NumPy arrays currently.

<details>

```python
In [1]: import numpy

In [2]: import numpy as np

In [3]: a = np.random.randint(0, 10, (5, 6))

In [4]: a
Out[4]: 
array([[2, 8, 5, 6, 9, 1],
       [2, 9, 8, 3, 2, 5],
       [1, 0, 4, 2, 5, 8],
       [6, 5, 1, 6, 8, 9],
       [6, 7, 6, 3, 7, 2]])

In [5]: np.asarray([a, a])
Out[5]: 
array([[[2, 8, 5, 6, 9, 1],
        [2, 9, 8, 3, 2, 5],
        [1, 0, 4, 2, 5, 8],
        [6, 5, 1, 6, 8, 9],
        [6, 7, 6, 3, 7, 2]],

       [[2, 8, 5, 6, 9, 1],
        [2, 9, 8, 3, 2, 5],
        [1, 0, 4, 2, 5, 8],
        [6, 5, 1, 6, 8, 9],
        [6, 7, 6, 3, 7, 2]]])
```

</details>